server/src/main/java/org/opensearch/gateway/remote/ClusterMetadataManifest.java

```diff
@@ -319,6 +391,8 @@ public void writeTo(StreamOutput out) throws IOException {
         if (out.getVersion().onOrAfter(Version.V_2_12_0)) {
             out.writeInt(codecVersion);
             out.writeString(globalMetadataFileName);
+        } else if (out.getVersion().onOrAfter(Version.V_2_14_0)) {
```

Comment:
We would need to change to `if` instead of `else if` so both blocks are executed for versions after 2.14